geode-serialization/src/main/java/org/apache/geode/internal/serialization/internal/AbstractSerializationContext.java

```diff
@@ -25,10 +26,10 @@
  */
 public abstract class AbstractSerializationContext {
 
-  <IO> Version getVersionForDataStream(final IO in) {
+  <IO> VersionOrdinal getVersionForDataStream(final IO in) {
     // check if this is a versioned data input
     if (in instanceof VersionedDataStream) {
-      final Version v = ((VersionedDataStream) in).getVersion();
+      final VersionOrdinal v = ((VersionedDataStream) in).getVersion();
       return v != null ? v : Version.getCurrentVersion();
```

Comment:
the return here may be either Version or VersionOrdinal, do we want to upcast or downcast to ensure returning a single type? Doesn't appear to be a problem currently though

Reply:
"fixed" this code by reverting it to deal exclusively with `Version` as before the PR